### ISSUE TITLE
yum module reporting success even when unable to download RPM

### ISSUE THREAD

**Issue by acranox** [ansible/ansible#6605]
##### Issue Type:

Bug Report
##### Ansible Version:

ansible 1.4.1
##### Environment:

CentOS 6.5
##### Summary:

installing an RPM with yum=http://url.com/foo.rpm reports "OK" even when the rpm was never downloaded due to a network error.
##### Steps To Reproduce:

I have a task that is essentially:

```
yum: name=http://www.example.com/packages.rpm state=present
```

and the client machine is unable to reach www.example.com, but the playbook still reported success.

```
TASK: [packages | install  packages] ************************
ok: [server1] => (item=http://www.example.com/packages.rpm)
```

a manual run looks like this:

```
[user@server1:~] 134$ sudo /usr/bin/yum -d 2 -y install http://www.example.com/packages.rpm
Setting up Install Process
Cannot open: http://www.example.com/packages.rpm. Skipping.
Error: Nothing to do

[user@server1:~] 135$ echo $?
1
```
##### Expected Results:

I expect the TASK to report a failure.
##### Actual Results:

the task reports success like this:

```
TASK: [packages | install  packages] ************************
ok: [server1] => (item=http://www.example.com/packages.rpm)
```

**Comment by mpdehaan** [comment]
Thanks for the report.

I suspect this is specific to http:// sources.

**Comment by jimi-c** [comment]
This appears to be resolved as of Ansible v1.5.0+. Could you please re-test with one of these newer versions to verify?

**Comment by jimi-c** [comment]
Since this has been fixed in more recent versions, I'm going to go ahead and close this. If you continue to experience problems regarding this issue, please let us know. Thanks!
